### PR TITLE
Feat/add gemma4 support

### PR DESCRIPTION
**Pull request by EvineR666**
# PR type
- [√] Bug Fix
- [ ] New Feature
- [ ] Document Updates
- [√] More Models or Datasets Support

# PR information
This PR mainly contains two changes:
1. Add full support for Google Gemma4 series models (require transformers>=5.8.0)
2. Fix the bug that Template and InputProcessor could not correctly process multimodal data in messages

## Experiment results
The training loss curve of the Gemma4-E2B model , dataset=Self-Cognition
2ad0b9312c9833fd9bb986335e5736ba

### CODE REVIEW

File: src/twinkle/template/gemma4.py
Comment (tpx818) [inline src/twinkle/template/gemma4.py:1]: 这个类是否有存在的必要

**Comment by tpx818** [comment]
请在readme中补充gemma4的支持

**Comment by tpx818** [comment]
附上正常训练的图

File: cookbook/mm/fsdp2_gemma4_mm.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+            'expert_parallel': {
+                'enabled': True,
+                'router_dtype': 'fp32',
+                'keep_router_logits': False,
+            }
+        },
+    )
+    # 若未传入config, 则自动通过 AutoConfig.from_pretrained 读取config
+
+    # model.model._no_split_modules = {'Gemma4VisionEncoderLayer', 'Gemma4TextDecoderLayer', 'Gemma4AudioLayer'}    # for 3 modalities(2B、4B)
```
Comment (tpx818) [inline cookbook/mm/fsdp2_gemma4_mm.py]: 无意义的注释，如果的确是不需要的，直接删除

File: src/twinkle/template/__init__.py [hunk 2]
```diff
@@ -3,3 +3,4 @@
 from .deepseek_v4 import DeepseekV4Template
 from .qwen import QwenTemplate
 from .qwen3_5_vl import Qwen3_5Template
+from .gemma4 import Gemma4Template
```
Comment (tpx818) [inline src/twinkle/template/__init__.py]: 这个引用不存在

**Review by tpx818 — APPROVED** [review]
